lib/ruboty/actions/jobcan.rb

```diff
@@ -23,48 +23,105 @@ def remember_group_id
         message.reply("I remember.")
       end
 
-      def punch_clock(in_out = :auto)
-        unless (code = user_data["code"])
+      def register_group_alias
+        user_data["alias_#{message[:group_name]}"] = message[:group_id]
+        message.reply("I registered alias #{message[:group_name]} to ID #{message[:group_id]}.")
+      end
+
+      def punch_clock(in_out = :auto, at: nil)
+        if !present_login_env? && !(code = user_data["code"])
           message.reply("I don't know your JOBCAN code.")
           return
+        else
+          code = nil
         end
-        unless (group_id = user_data["group_id"])
-          message.reply("I don't know your JOBCAN group ID.")
-          return
-        end
+
+        group_id = if at
+                     user_data["alias_#{at}"]
+                   elsif !user_data["group_id"].nil?
+                     user_data["group_id"]
+                   else
+                     message.reply("I don't know your JOBCAN group ID.")
+                   end
+
         client = JobcanClient.new(code, group_id, in_out)
-        client.authenticate!
+        client.authenticate!(post: present_login_env?)
         status = client.punch_clock!
         message.reply("OK, your current status is #{status}.")
       rescue
         message.reply("Error: #{$!}")
       end
 
+      def punch_clock_at
+        unless exist_group_alias?(message[:group_name])
+          cannot_find_alias_message(message[:group_name])
+          return
+        end
+        punch_clock(:auto, at: message[:group_name])
+      end
```

Comment:
I would simply pass the given group name to `punch_clock`

```ruby
def punch_clock_at
  punch_clock(:auto, at: message[:group_name])
end
```

and let `puch_clock` resolve alias

```ruby
def punch_clock(in_out = :auto, at = nil)
  ...
  group_id = if at
               user_data["alias_#{at}"] ||
                 return message.reply("I don't know your JOBCAN group alias '#{at}'.")
             else
               user_data["group_id"] ||
                 return message.reply("I don't know your JOBCAN group ID.")
             end
  ...
end
```